presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -382,18 +388,11 @@ public boolean trySetMemoryReservation(long newMemoryReservation)
     {
         checkArgument(newMemoryReservation >= 0, "newMemoryReservation is negative");
 
-        long delta = newMemoryReservation - memoryReservation.get();
-
+        long delta = newMemoryReservation - operatorMemoryContext.getUserMemory();
```

Comment:
I see that you are trying to get a `delta` value here. Why do we have `tryUpdateUserMemory` (instead of `trySetUserMemory`) on `MemoryTrackingContext`?